libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConfig.java

```diff
@@ -234,6 +249,7 @@ public Builder() {
             outgoingMsgSeqNum = 1;
             maxFieldCount     = 64;
             fieldCapacity     = 64;
+            blockSize         = 4096;
```

Comment:
The memory layout of Java arrays in memory is as follows: (1) object header (16 bytes) and (2) the array elements.

IOW, if you want to be fancy and micro-optimize, make the `blockSize` be 4080 by default, and your array object will fit in one 4 KiB hardware page:

```java
package parity;

import org.openjdk.jol.info.ClassLayout;

public class MemoryLayout {
  public static void main(String[] args) {
    System.out.println(ClassLayout.parseInstance(new byte[4096]).toPrintable());
    System.out.println(ClassLayout.parseInstance(new byte[4080]).toPrintable());
  }
}
```

```console
# WARNING: Unable to get Instrumentation. Dynamic Attach failed. You may add this JAR as -javaagent manually, or supply -Djdk.attach.allowAttachSelf
[B object internals:
OFF  SZ   TYPE DESCRIPTION               VALUE
  0   8        (object header: mark)     0x0000000000000001 (non-biasable; age: 0)
  8   4        (object header: class)    0x00076862
 12   4        (array length)            4096
 12   4        (alignment/padding gap)
 16 4096   byte [B.<elements>             N/A
Instance size: 4112 bytes
Space losses: 4 bytes internal + 0 bytes external = 4 bytes total

[B object internals:
OFF  SZ   TYPE DESCRIPTION               VALUE
  0   8        (object header: mark)     0x0000000000000001 (non-biasable; age: 0)
  8   4        (object header: class)    0x00076862
 12   4        (array length)            4080
 12   4        (alignment/padding gap)
 16 4080   byte [B.<elements>             N/A
Instance size: 4096 bytes
Space losses: 4 bytes internal + 0 bytes external = 4 bytes total
```

Reply:
This is a good point, @penberg! 👍🏼  I'm actually in the process of rebasing this pull request on `master` again; some of the `FIXValue` internals have got changed since the spring of 2020.